packages/did-provider-jwk/src/resolver.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  DIDDocument,
+  DIDResolutionOptions,
+  DIDResolutionResult,
+  DIDResolver,
+  ParsedDID,
+  Resolvable,
+  JsonWebKey,
+} from 'did-resolver'
+import { encodeBase64url, decodeBase64url } from '@veramo/utils'
+
+const isJWK = (data: unknown): data is JsonWebKey => {
+  if (
+    typeof data === 'object' &&
+    data &&
+    'crv' in data &&
+    typeof data.crv === 'string' &&
+    'kty' in data &&
+    'x' in data &&
+    typeof data.x === 'string' &&
+    ((data.kty === 'EC' && 'y' in data && typeof data.y === 'string') ||
+      (data.kty === 'OKP' && !('y' in data)))
+  ) {
+    return true
+  }
+  return false
+}
+
+function generateDidDocument(jwk: JsonWebKey): Promise<DIDDocument> {
+  return new Promise((resolve, reject) => {
+    try {
+      const did = `did:jwk:${encodeBase64url(JSON.stringify(jwk))}`
+      const didDocument: DIDDocument = {
+        id: did,
+        '@context': ['https://www.w3.org/ns/did/v1', 'https://w3id.org/security/suites/jws-2020/v1'],
+        verificationMethod: [
+          {
+            id: `${did}#0`,
+            type: 'JsonWebKey2020',
+            controller: did,
+            publicKeyJwk: jwk,
+          },
+        ],
+        assertionMethod: [`${did}#0`],
+        authentication: [`${did}#0`],
+        capabilityInvocation: [`${did}#0`],
+        capabilityDelegation: [`${did}#0`],
+        keyAgreement: [`${did}#0`],

```

Comment:
According to the did:jwk spec, the key relationships should consider the `use` property of a JWK.
* When `use` is `enc`, only the `keyAgreement` relationship should be added.
* When `use` is `sig`, all except the `keyAgreement` relationship should be added.

I'm not aware of other implementations of this resolver so I can't tell if imposing this alignment with the spec would cause incompatibilities. What do you think?

Reply:
I have added the `use` property to the resolved JWK object and the corresponding key relationships according to the spec.